### PR TITLE
[fix](multi-catalog)add the max compute fe ut and fix download expired

### PR DESCRIPTION
**Pull request by wsjz**
## Proposed changes

1. add the max compute fe ut and fix download expired
2. solve memery leak when allocator close
3. add correct partition rows

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java [hunk 1]
```diff
@@ -167,10 +172,21 @@ public void open() throws IOException {
...
             maxComputeColumns.removeIf(e -> partitionColumns.contains(e.getName()));
             curReader = session.openArrowRecordReader(start, totalRows, maxComputeColumns, arrowAllocator);
+        } catch (TunnelException e) {
+            if (retryCount > 0 && e.getErrorMsg().contains("TableModified")) {
+                retryCount--;
+                // try to refresh table scan and re-open odps
+                refreshTableScan();
+                open();
+            } else {
+                retryCount = 2;
```
Comment (morningman) [inline fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:182]: Why need to reset `retryCount`?
I thought the MaxComputeJniScanner will be new for each query?

Comment (wsjz) [reply fe/be-java-extensions/max-compute-scanner/src/main/java/org/apache/doris/maxcompute/MaxComputeJniScanner.java:182]: avoid to call open twice

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run p0

**Comment by wsjz** [comment]
run external

**Review by morningman — APPROVED** [review]
LGTM

**Comment by morningman** [comment]
run feut beut clickbench

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run clickbench

**Comment by morningman** [comment]
run beut

**Review by kaka11chen — APPROVED** [review]
LGTM

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Comment by wsjz** [comment]
run buildall

**Review by zy-kkk — APPROVED** [review]

**Pull request merged**